src/components/link/link.helper.js

```diff
@@ -57,22 +58,60 @@ function getRadiusStrategy(type) {
     return RADIUS_STRATEGIES[type] || RADIUS_STRATEGIES[LINE_TYPES.STRAIGHT];
 }
 
+const cacheLinksCount = new Map();
+const cacheIdCount = new Map();
+const _key = (s, t) => `${s}:${t}`;
+const addLink = (s, t, id) => {
```

Comment:
- [ ] This cache implementation is messy, I would definitely get this refactored moving forward.